.github/workflows/test_pull_requests.yml

```
@@ -187,25 +187,51 @@ jobs:
     name: test benchmarks
     runs-on: ubuntu-latest
     timeout-minutes: 60
+    env:
+      GIT_LFS_SKIP_SMUDGE: 1
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
       - uses: actions/setup-python@v5
         with:
           python-version: 3.11
           cache-dependency-path: setup.cfg
 
       - uses: tlambert03/setup-qt-libs@v1
 
+      - uses: octokit/request-action@v2.x
+        # here we get hash of the latest release commit to compare with PR
+        id: latest_release
+        with:
+          route: GET /repos/{owner}/{repo}/releases/latest
+          owner: napari
+          repo: napari
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
       - name: install dependencies
         run: |
           pip install --upgrade pip
           pip install asv[virtualenv]
 
-      - name: Run benchmarks
-        uses: aganders3/headless-gui@v2
+      - name: asv machine
+        run: asv machine --yes
+
+      - name: Run benchmarks PR
+        uses: aganders3/headless-gui@v1
         with:
           run: |
-            asv machine --yes
             asv run --show-stderr --quick  --attribute timeout=300 HEAD^!
         env:
           PR: 1
+
+      - name: Run benchmarks latest release
+        # here we check if the benchmark on the latest release is not broken
+        uses: aganders3/headless-gui@v1
+        with:
+          run: |
+            asv run --show-stderr --quick  --attribute timeout=300  ${{ fromJSON(steps.latest_release.outputs.data).target_commitish }}^!
+        env:
+          PR: 1
```

1?

I'd love some guidance here, perhaps in a comment, about what to do if the benchmarks break in your PR because of an API change...

---

`${{ fromJSON(steps.latest_release.outputs.data).target_commitish }}` is getting output of `latest_release` steep and extract from it hash of release commit.

`1` is same as in `Run benchmarks PR` steep. And is used to indicate to omit long computing case (and reduce time from 10h to 6 minutes).

---

> And is used to indicate to omit long computing case (and reduce time from 10h to 6 minutes).

Maybe worth adding a comment to those lines? 

```suggestion
          PR: 1  # prevents asv from running very compute-intensive benchmarks
```